src/client/Client.h

```diff
@@ -534,6 +534,8 @@ class Client : public Dispatcher, public md_config_obs_t {
    */
   void _handle_full_flag(int64_t pool);
 
+  void close_sessions();
```

Comment:
Yeah, it's really more around our interface functions than everywhere. But since we already used this one the wrong way once, and it's not clear from local context that you need the client lock, that seems like the easiest indicator.